satpy/tests/reader_tests/test_generic_image.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/python
+# Copyright (c) 2018.
+#
+
+# Author(s):
+#   Panu Lahtinen <panu.lahtinen@fmi.fi>
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""Unittests for generic image reader.
+"""
+
+import os
+import unittest
+
+import xarray as xr
+import dask.array as da
+import numpy as np
+
+from satpy.scene import Scene
+from satpy.readers.generic_image import GenericImageFileHandler
+from satpy.readers.generic_image import get_geotiff_area_def
+from satpy.readers.generic_image import mask_image_data
+from satpy import CHUNK_SIZE
```

Comment:
Could you move all satpy-related imports to inside the tests so that import errors don't cause all tests to fail? @mraspaud was the first person to point this out to me.